test/Constraints/init_literal_via_coercion.swift

```diff
@@ -0,0 +1,22 @@
+// RUN: %target-swift-frontend %s -typecheck -debug-constraints -swift-version 4 2>%t.err
+// RUN: %FileCheck %s < %t.err
+
+// SE-0213. `UInt32(0)` and similar expressions that get transformed into
+// `0 as <#Type#>` should get literal bound early via equality constraint.
+
+// CHECK: ---Constraint solving at [{{.*}}:12:1 - line:12:13]---
+// CHECK: (integer_literal_expr type='[[LITERAL_VAR:\$T[0-9]+]]' {{.*}}
+// CHECK: Type Variables:
+// CHECK: [[LITERAL_VAR]] [allows bindings to: {{.*}}] as UInt32 {{.*}}
+// CHECK-NOT: disjunction (remembered) \[\[locator@{{.*}} [Coerce@{{.*}}\]\]]:
+_ = UInt32(0)
```

Comment:
Note that this could have been a unit test but unfortunately `CoerceExpr::forLiteralInit` depends on valid source locations...